redbot/cogs/economy/economy.py

```diff
@@ -127,7 +127,7 @@ class Economy(commands.Cog):
 
     default_global_settings = default_guild_settings
 
-    default_member_settings = {"next_payday": 0, "last_slot": 0}
+    default_member_settings = {"last_payday": 0, "last_slot": 0}
```

Comment:
Dont change the Setting key please .. unless you are going to write logic to migrate users with `next_payday` 

Keep it as `next_payday` and change your`await self.config.user(author).last_payday()` to `await self.config.user(author).next_payday()`